beta-operator-rewards/rewards.ts

```diff
@@ -0,0 +1,693 @@
+      .concat(intervalAuthorizationDecreasedEvents)
+      .filter((event) => event.args.stakingProvider === stakingProvider);
+
+    /// Random Beacon application authorization requirement
+    let beaconAuthorization = await getAuthorization(
+      randomBeacon,
+      intervalEvents,
+      postIntervalIncreasedEvents.concat(postIntervalDecreasedEvents),
+      stakingProvider,
+      startRewardsBlock,
+      endRewardsBlock,
+      currentBlockNumber
+    );
+    authorizations.set(BEACON_AUTHORIZATION, beaconAuthorization.toString());
+    requirements.set(IS_BEACON_AUTHORIZED, !beaconAuthorization.isZero());
+
+    /// tBTC application authorized requirement
+    const tbtcAuthorization = await getAuthorization(
+      walletRegistry,
+      intervalEvents,
+      postIntervalIncreasedEvents.concat(postIntervalDecreasedEvents),
+      stakingProvider,
+      startRewardsBlock,
+      endRewardsBlock,
+      currentBlockNumber
+    );
+
+    authorizations.set(TBTC_AUTHORIZATION, tbtcAuthorization.toString());
+    requirements.set(IS_TBTC_AUTHORIZED, !tbtcAuthorization.isZero());
+
+    /// Off-chain client reqs
+
+    // Populate instances for a given operator.
+    await instancesForOperator(operatorAddress, rewardsInterval, instancesData);
+
+    /// Uptime requirement
+    let { uptimeCoefficient, isUptimeSatisfied } = await checkUptime(
+      operatorAddress,
+      rewardsInterval,
+      instancesData
+    );
+    // BigNumbers cannot operate on floats. Coefficient needs to be multiplied by 100
+    uptimeCoefficient = Math.floor(uptimeCoefficient * HUNDRED);
+    requirements.set(IS_UP_TIME_SATISFIED, isUptimeSatisfied);
+
+    /// Pre-params requiremnt
+    const isPrePramsSatisfied = await checkPreParams(
+      operatorAddress,
+      rewardsInterval,
+      instancesData
+    );
+
+    requirements.set(IS_PRE_PARAMS_SATISFIED, isPrePramsSatisfied);
+
+    /// Version requirement
+    let {
+      instanceWithLatestBuildVersion,
+      latestRegisteredBuildVersionTimestmap,
+    } = await checkVersion(operatorAddress, rewardsInterval, instancesData);
+
+    if (instanceWithLatestBuildVersion === undefined) {
+      console.log(
+        `Cannot determine a client version. No Rewards were calculated for operator ${operatorAddress}`
+      );
+      continue;
+    }
+
+    // This is an example to illustrate a client's build version requirement.
+    // A client must be either on the second to latest version or latest.
+    //                       v1 or v2                v2
+    //           |-------------------------------\|-------|
+    // Timeline -|-------------*------------------*-------|->
+    //         Sep1            v2             v2+delay   Sep30
+    // Where:
+    // v2 was released Sep10
+    // delay = 14days
+    // v2 + delay = Sep10 + 14days = Sep24
+    // Between Sep1 - Sep24 a client is allowed to run v1 or v2
+    // Between Sep24 - Sep30 a client is allowed to run only v2
```

Comment:
Can you move the version validation logic out to its own function? I think it's complex enough that it deserves its own call